core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -49,7 +49,7 @@ class AdminClient(val time: Time,
 
     do {
       future = client.send(target, api, request)
-      client.poll(future)
+      client.poll(future, deadline - now)
```

Comment:
Maybe we should add a check when this returns to see if the future is finished? If it's not finished, future.exception() will probably be null, and maybe we can throw a TimeoutException.